Importlib's PathFinder has an incorrect interface definition

The [`importlib.machinery.PathFinder`](https://github.com/python/typeshed/blame/256b3ce8aba6080f9a1c90c8c264e93e49e754cd/stdlib/3/importlib/machinery.pyi#L84) stub indicates that it has the same interface as `importlib.abc.MetaPathFinder`. This is incorrect because the `find_module()`, `invalidate_caches()`, and `find_spec()` methods are class methods in `PathFinder` and instance methods in `MetaPathFinder`. It also has additional `_path_hooks()`, `_path_importer_cache`, `_legacy_get_spec()`, `_get_spec()`, and `find_distributions()` methods that aren't included in the type definition.

A quick reproducible example:

```python
from importlib.machinery import PathFinder

PathFinder.find_module('exodus')
```

results in:

```
t.py:3: error: Too few arguments for "find_module" of "MetaPathFinder"
t.py:3: error: Argument 1 to "find_module" of "MetaPathFinder" has incompatible type "str"; expected "MetaPathFinder"
Found 2 errors in 1 file (checked 1 source file)
```

--- a/stdlib/3/importlib/machinery.pyi
+++ b/stdlib/3/importlib/machinery.pyi
@@ -1,10 +1,10 @@
 import importlib.abc
 import types
-from typing import Any, Callable, List, Optional, Sequence, Tuple
+from typing import Any, Callable, List, Optional, Sequence, Tuple, Union
 
 # ModuleSpec is exported from this module, but for circular import
 # reasons exists in its own stub file (with Loader and ModuleType).
-from _importlib_modulespec import ModuleSpec as ModuleSpec  # Exported
+from _importlib_modulespec import ModuleSpec as ModuleSpec, Loader  # Exported
 
 class BuiltinImporter(importlib.abc.MetaPathFinder,
                       importlib.abc.InspectLoader):
@@ -81,7 +81,17 @@ class WindowsRegistryFinder(importlib.abc.MetaPathFinder):
                   target: Optional[types.ModuleType] = ...) -> Optional[ModuleSpec]:
         ...
 
-class PathFinder(importlib.abc.MetaPathFinder): ...
+class PathFinder:
+    @classmethod
+    def invalidate_caches(cls) -> None: ...
+    @classmethod
+    def find_spec(cls, fullname: str,
+                  path: Optional[Sequence[Union[bytes, str]]] = ...,
+                  target: Optional[types.ModuleType] = ...) -> Optional[ModuleSpec]: ...
+    @classmethod
+    def find_module(cls, fullname: str,
+                    path: Optional[Sequence[Union[bytes, str]]] = ...) -> Optional[Loader]: ...
+
 
 SOURCE_SUFFIXES: List[str]
 DEBUG_BYTECODE_SUFFIXES: List[str]

--- a/tests/stubtest_whitelists/py3_common.txt
+++ b/tests/stubtest_whitelists/py3_common.txt
@@ -418,9 +418,6 @@ importlib.machinery.ExtensionFileLoader.get_filename
 importlib.machinery.FrozenImporter.find_module
 importlib.machinery.FrozenImporter.find_spec
 importlib.machinery.FrozenImporter.module_repr
-importlib.machinery.PathFinder.find_module
-importlib.machinery.PathFinder.find_spec
-importlib.machinery.PathFinder.invalidate_caches
 importlib.machinery.SourceFileLoader.set_data
 importlib.machinery.WindowsRegistryFinder.find_module
 importlib.machinery.WindowsRegistryFinder.find_spec

